pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -85,15 +91,21 @@ func (g *Gatherer) GetName() string {
 // + the gathering function producing metadata for the conditional gatherer
 func (g *Gatherer) GetGatheringFunctions(ctx context.Context) (map[string]gatherers.GatheringClosure, error) {
 	newGatheringRules, err := g.fetchGatheringRulesFromServer(ctx)
-	klog.Infof("got %v gathering rules for conditional gatherer", len(newGatheringRules))
+	klog.Infof(
+		"got %v gathering rules for conditional gatherer with version %v",
+		len(newGatheringRules.Rules), newGatheringRules.Version,
+	)
 	if err != nil {
 		klog.Errorf("unable to fetch gathering rules from the server: %v", err)
-		klog.Infof("trying to use cached gathering config containing %v gathering rules", len(g.gatheringRules))
+		klog.Infof(
+			"trying to use cached gathering config containing %v gathering rules and version %v",
+			len(g.gatheringRules.Rules), g.gatheringRules.Version,
+		)
```

Comment:
This is only idea. I am thinking if it would be slightly more clear to move the rest of the `GetGatheringFunctions` to some new function (e.g `createGatheringFunctions`) and return here with the function call `return createGatheringFunctions(g.gatheringRules)` and remove the else and simply return `return createGatheringFunctions(newGatheringRules)` then. And perhaps return nil in https://github.com/openshift/insights-operator/pull/601/files#diff-57e9d48447e382f53bafe213301e653fd5989c354c9d751e7f6386be25459e89R169.

Reply:
done